libfreerdp/core/gateway/arm.c

```diff
@@ -956,6 +956,10 @@ BOOL arm_resolve_endpoint(rdpContext* context, DWORD timeout)
 				WLog_DBG(TAG, "Delay for %" PRIdz "ms before next attempt", delay);
 				Sleep(delay);
 			}
+			else
```

Comment:
I'd suggest inverting the if condition 
```C
if (delay == 0)
   break;

WLog_DBG(TAG, "Delay for %" PRIdz "ms before next attempt", delay);
Sleep(delay);
```

Reply:
hmm, I´d say preference. and would be `delay <= 0` as `-1` is returned in error cases.